pkg/coscheduling/coscheduling.go

```diff
@@ -304,3 +342,59 @@ func (cs *Coscheduling) calculateWaitingPods(podGroupName, namespace string) int
 
 	return waiting
 }
+
+func (cs *Coscheduling) cleanPodGroupInfoIfPresent(obj interface{}) {
+	var pod *v1.Pod
+	switch t := obj.(type) {
+	case *v1.Pod:
+		pod = t
+	case cache.DeletedFinalStateUnknown:
+		var ok bool
+		pod, ok = t.Obj.(*v1.Pod)
+		if !ok {
+			klog.Errorf("cannot convert to *v1.Pod: %v", t.Obj)
+			return
+		}
+	default:
+		klog.Errorf("cannot convert to *v1.Pod: %v", t)
+		return
+	}
+	cs.deletePodGroupInfo(pod)
+}
+
+// assignedPod selects pods that are assigned (scheduled and running).
+func assignedPod(pod *v1.Pod) bool {
+	return len(pod.Spec.NodeName) != 0
+}
+
+func (cs *Coscheduling) deletePodGroupInfo(pod *v1.Pod) {
+	podGroupName, podMinAvailable, _ := GetPodGroupLabels(pod)
+	if len(podGroupName) > 0 && podMinAvailable > 0 {
+		pgKey := fmt.Sprintf("%v/%v", pod.Namespace, podGroupName)
+		// If it is a PodGroup and present in PodGroupInfos, set it's deleted true.
+		if len(pgKey) != 0 {
+			value, exist := cs.podGroupInfos.Load(pgKey)
+			if exist {
+				pgInfo := value.(*PodGroupInfo)
+				if !pgInfo.deleted {
+					pgInfo.deleteTime = time.Now()
+				}
```

Comment:
It marks the PodGroup as deletable when its first Pod gets deleted, is it a reasonable assumption?

Reply:
We have two choice to mark the PodGroup as deletable. 1. when its first pod gets deleted. 2 when the all the pods of PodGroup are deleted.
The second will lead more frequent operations of sync.map because we will store the pod in PodGroupInfo, it will degrade performance much. So we choose the set the PodGroup deleted. If some pods of PodGroup still left, we will remove the deleted status in 
```go
value, exist := cs.podGroupInfos.Load(pgKey)
		if exist {
			pgInfo := value.(*PodGroupInfo)
			if pgInfo.deletionTimestamp != nil {
				pgInfo.deletionTimestamp = nil
				cs.podGroupInfos.Store(pgKey, pgInfo)
			}
			return pgInfo, podMinAvailable
		}
```

Reply:
> Thanks!
> 
> /lgtm
> /approve
> /hold
> in case @yuanchen8911 wants to take a final look.

lgtm to me too. @Huang-Wei